Data/Vector/Generic.hs

```diff
@@ -432,6 +434,14 @@ splitAt n v = ( unsafeSlice 0 m v
       n'  = max n 0
       len = length v
 
+-- | /O(1)/ Yield the 'head' and 'tail' of the vector, or 'Nothing' if empty.
+uncons :: Vector v a => v a -> Maybe (a, v a)
+uncons xs = flip (,) (tail xs) <$> xs !? 0
```

Comment:
Wouldn't it be more efficient to use `unsafeTail` here, since you know in this case that the `Vector` is non-empty?